### PR TITLE
Add tests for KeyringController edge cases

### LINKED ISSUE #3767: Bump `KeyringController` test coverage to 100%
After merging `EthKeyringController` into `KeyringController`, we should strive to keep the test coverage to a minimum of 100%, and ensure that all the test cases still make sense after the increase in responsibilities on the resulting single `KeyringController`.

This issue is part of this original plan: #2043

### PR DESCRIPTION
**Pull request by mikesposito**
## Explanation

This PR adds additional tests for edge cases, useful for #3830.
Some `istanbul ignore if` statements are also being removed

## References

* Fixes #3767 
* See #3830 

## Changelog

No functional changes

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/keyring-controller/src/KeyringController.test.ts [hunk 1]
```diff
@@ -2762,7 +3036,9 @@ type WithControllerCallback<ReturnValue> = ({
   messenger: KeyringControllerMessenger;
 }) => Promise<ReturnValue> | ReturnValue;
 
-type WithControllerOptions = Partial<KeyringControllerOptions>;
+type WithControllerOptions = Partial<KeyringControllerOptions> & {
```
Comment (mcmire) [inline packages/keyring-controller/src/KeyringController.test.ts:3039]: Nit: Something I've started doing in other test files when it comes to `withController` (or other similar helpers, like `setupController`) is splitting off controller options from other kinds of options. So `withController` could take an options object, but the controller options could be bundled into an `options` object, and then any other setup configuration you want to make could be done in other options. Essentially that turns this type into:

``` typescript
type WithControllerOptions = {
  options: Partial<KeyringControllerOptions>;
  skipVaultCreation?: boolean;
}
```

I realize that this would require a change across the board, but something to think about perhaps for a future PR.

**Review by mcmire — APPROVED** [review]
Looks good!

File: packages/keyring-controller/src/KeyringController.test.ts (hunk 1 above)
Comment (mikesposito) [reply packages/keyring-controller/src/KeyringController.test.ts:3109]: Good idea, I do think it would improve readability! I will do that in another PR
